src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -184,6 +184,7 @@ def put(
         Update a member's organization and team-level roles.
         """
         allowed_roles = get_allowed_org_roles(request, organization)
+        allowed_old_roles = get_allowed_org_roles(request, organization, include_retired=True)
```

Comment:
Ideally I'd prefer to make only one call to `get_allowed_org_roles` and rely on the `Role.is_retired` field to distinguish between the two cases afterward. Not for efficiency's sake, but to avoid the cognitive overhead of two sets of roles.

I have a few ideas about how to refactor this, but it's not worth holding up the fix over. But I might make a quick pass after this is merged?

Reply:
+1